frontend/packages/kubevirt-plugin/src/selectors/vm-template/advanced.ts

```diff
@@ -49,6 +56,19 @@ export const getTemplatesLabelValues = (templates: TemplateKind[], label: string
 export const getTemplateFlavors = (templates: TemplateKind[]) =>
   getTemplatesLabelValues(templates, TEMPLATE_FLAVOR_LABEL);
 
+export const getTemplateHostname = (template: TemplateKind) => {
```

Comment:
IMO, this selector could be a bit confusing to have in this place. Can we limit this selector to just this line ?

```tsx
_.get(vm, 'spec.template.spec') && vm.spec.template.spec.hostname;
```

and move the rest as a new function to either prefill-vm-template-state-update.ts or some other location?

Reply:
IMO its in place, what is confusing here?
As a dev I want to know the hostname either from spec, if not try cloudinit, and if not return undefined.

Reply:
okay,  I guess we can keep it together until we need that separately